### PR TITLE
LiveIntent User ID Module: Eliminating live-connect NPM Dependency

### PR DESCRIPTION
**Pull request by 3link**
## Type of change
- [x] Feature

## Description of change
Allows to use the LiveIntent user ID module with an instance of LiveConnect v3 present on the page instead of inlining the LiveConnect NPM module into the user ID module as described here: https://github.com/prebid/Prebid.js/issues/11541.

The goal is to support both `standard`/`minimal` mode as well as the `external` mode during the migration phase. Once all our customers have migrated away from using `standard`/`minimal` mode, we will deprecate this mode and delete the LiveConnect NPM module dependency from Prebid together with all the code that depends on it. That would include moving the code from `libraries/liveIntentId` back into `modules/liveintentIdSystem` as we would not need to differentiate between two module implementations.

## Other information
- Resolves https://github.com/prebid/Prebid.js/issues/11541
- Documentation changes: https://github.com/prebid/prebid.github.io/pull/5567

### CODE REVIEW

**Comment by 3link** [comment]
@jlquaccia I have a question about possible ways to extend our user id module that depends on the external LiveConnect. Would it be possible to extend the module configuration such that a URL can be provided from which to download the LiveConnect script and the user id module would then embed this script on the page? If no such URL is provided in the configuration, nothing will be embedded.

That would make the switch for our customers smoother as they will just need to add an additional config to our user id module and it will automatically deploy the script.

**Comment by jlquaccia** [comment]
> @jlquaccia I have a question about possible ways to extend our user id module that depends on the external LiveConnect. Would it be possible to extend the module configuration such that a URL can be provided from which to download the LiveConnect script and the user id module would then embed this script on the page? If no such URL is provided in the configuration, nothing will be embedded.
> 
> That would make the switch for our customers smoother as they will just need to add an additional config to our user id module and it will automatically deploy the script.

Overall, I think the changes on this PR seem fine to me.  In terms of allowing a URL to be provided via module configuration, I'm not sure if that is a direction that Prebid would like to go in since it seems it could open some security related scenarios.  @ncolletti @patmmccann what are your thoughts

**Review by jlquaccia — APPROVED** [review]

**Comment by 3link** [comment]
> > @jlquaccia I have a question about possible ways to extend our user id module that depends on the external LiveConnect. Would it be possible to extend the module configuration such that a URL can be provided from which to download the LiveConnect script and the user id module would then embed this script on the page? If no such URL is provided in the configuration, nothing will be embedded.
> > That would make the switch for our customers smoother as they will just need to add an additional config to our user id module and it will automatically deploy the script.
> 
> Overall, I think the changes on this PR seem fine to me. In terms of allowing a URL to be provided via module configuration, I'm not sure if that is a direction that Prebid would like to go in since it seems it could open some security related scenarios. @ncolletti @patmmccann what are your thoughts

Just to provide more context here: our customers would now be required to place our LiveConnect script on their domains in order for our user ID sub-module to work. We have customers with thousands of domains and it would make upgrading to the newer Prebid/LiveIntent sub-module version much easier for them if instead of setting up LiveConnect on each of their pages manually, they could configure their URL for the LiveConnect script directly in our user ID module.

File: libraries/liveIntentId/externalIdSystem.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  if (cachedClientRef != null) return cachedClientRef
+
+  const clientRef = {}
+
+  const clientDetails = { name: 'prebid', version: '$prebid.version$' }
+
+  const collectConfig = configParams.liCollectConfig ?? {};
+
+  let integration
+  if (collectConfig.appId != null) {
```
Comment (ncolletti) [inline libraries/liveIntentId/externalIdSystem.js:29]: is it worth checking for the optional `publisherId` param here?

Comment (3link) [reply libraries/liveIntentId/externalIdSystem.js:29]: @ncolletti The code here determines the type of the integration that depends on a combination of certain fields. In this case if the appId is provided, the actual value of the publisherId can be defined or undefined. Both is fine.

**Comment by ncolletti** [comment]
@3link As far as I'm aware Prebid has been upfront about not allowing external code to be loaded other than from RTD Modules where a contract/registration to acknowledge the contents of the JS by the publisher is signed prior. While cumbersome, I think this configuration is preferred by Prebid. 

Somewhat recent guidance on this topic: https://github.com/prebid/Prebid.js/issues/9570

Any case, I don't know if a change should happen in this iteration of your adapter. It's best to bring this up to the Identity PMC group for further discussion

**Comment by 3link** [comment]
> As far as I'm aware Prebid has been upfront about not allowing external code to be loaded other than from RTD Modules where a contract/registration to acknowledge the contents of the JS by the publisher is signed prior. While cumbersome, I think this configuration is preferred by Prebid.

> Somewhat recent guidance on this topic: https://github.com/prebid/Prebid.js/issues/9570

> Any case, I don't know if a change should happen in this iteration of your adapter. It's best to bring this up to the Identity PMC group for further discussion

@ncolletti That makes sense. Thank you for the link! 

I think it makes sense to keep these two topics separate and merge this version of the module as is for now.

**Review by ncolletti — APPROVED** [review]

**Comment by ncolletti** [comment]
@jlquaccia looks good to me, will let you give a final sign off and merge

**Comment by jlquaccia** [comment]
@ncolletti yep, looks good to me too.  Merging now..

File: libraries/liveIntentId/idSystem.js [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+ * @requires module:modules/userId
+ */
+import { triggerPixel, logError } from '../../src/utils.js';
+import { ajaxBuilder } from '../../src/ajax.js';
+import { gdprDataHandler, uspDataHandler, gppDataHandler } from '../../src/adapterManager.js';
+import { submodule } from '../../src/hook.js';
+import { LiveConnect } from 'live-connect-js'; // eslint-disable-line prebid/validate-imports
+import { getStorageManager } from '../../src/storageManager.js';
+import { MODULE_TYPE_UID } from '../../src/activities/modules.js';
+import { DEFAULT_AJAX_TIMEOUT, MODULE_NAME, composeIdObject, eids, DEFAULT_DELAY, GVLID, PRIMARY_IDS, parseRequestedAttributes } from './shared.js'
```
Comment (s-solodovnikov) [inline libraries/liveIntentId/idSystem.js:14]: Hi. You are importing the `DEFAULT_DELAY` const but it does not exist.

Comment (3link) [reply libraries/liveIntentId/idSystem.js:14]: https://github.com/prebid/Prebid.js/pull/12334

**Pull request merged**